deploy/libproject/deploy.rb

```diff
@@ -88,90 +162,87 @@ def deploy(cmd_name, args)
-        --app_version #{op.opts.app_version}
-        --git_version #{op.opts.git_version}
-        --key_file #{DOCKER_KEY_FILE_PATH}
-      }
-      return
-    end
+  unless Workbench::in_docker?
+    return setup_and_enter_docker(cmd_name, op.opts)
   end
 
   # Everything following runs only within Docker.
+  # Only require Jira stuff within Docker to avoid burdening the user with local
+  # workstation Ruby gem setup.
+  require_relative 'jirarelease'
+
   if op.opts.key_file.nil?
-    raise ArgumentError.new("--key_file is required when running within docker")
+    raise ArgumentError.new("--key-file is required when running within docker")
   end
   if op.opts.app_version.nil?
-    raise ArgumentError.new("--app_version is required when running within docker")
+    raise ArgumentError.new("--app-version is required when running within docker")
   end
   if op.opts.git_version.nil?
-    raise ArgumentError.new("--git_version is required when running within docker")
+    raise ArgumentError.new("--git-version is required when running within docker")
   end
+  common = Common.new
   common.run_inline %W{gcloud auth activate-service-account -q --key-file #{op.opts.key_file}}
 
-  # TODO: Create/update the Jira ticket.
+  jira_client = nil
+  create_ticket = false
+  from_version = nil
+  maybe_log_jira = lambda { |msg| common.status msg }
+  if op.opts.update_jira
+    if not VERSION_RE.match(op.opts.app_version) or
+      op.opts.app_version != op.opts.git_version
+      raise RuntimeError.new "for releases, the --git_version and " +
+                             "--app_version should be equal and should be a " +
+                             "release tag (e.g. v0-1-rc1); you shouldn't " +
+                             "bypass this, but if you need to you can pass " +
+                             "--no-update-jira"
+    end
+
+    # We're either creating a new ticket (staging), or commenting on an existing
+    # release ticket (stable, prod).
+    jira_client = JiraReleaseClient.from_gcs_creds(op.opts.project)
+    if op.opts.update_jira and op.opts.project == STAGING_PROJECT
+      create_ticket = true
+      from_version = get_live_gae_version(STAGING_PROJECT)
+      if not from_version
+        # Alternatively, we could support a --from_version flag
+        raise RuntimeError "could not determine live staging version, and " +
+                           "therefore could not generate a delta commit log; " +
+                           "please manually deploy staging with the old " +
+                           "version and supply --no-update-jira, then retry"
+      end
+    else
+      maybe_log_jira = lambda { |msg|
+        begin
+          jira_client.comment_ticket(op.opts.app_version, msg)
+        rescue StandardError => e
+          common.error "comment_ticket failed: #{e}"
+        end
+      }
+    end
+  end
+
+  maybe_log_jira.call "'#{op.opts.project}': Beginning deploy of api and " +
```

Comment:
This seems OK for now... but ideally we'd have a log of steps (update db, config, api, public-api, etc.) completed in JIRA, so that if the deploy fails midway we'll know how far we got. Maybe add a TODO to support that in future?

Reply:
Yep, added TODO.